qubes/tests/integ/vm_qrexec_gui.py

```diff
@@ -664,6 +664,34 @@ def test_110_qrexec_filecopy_deny(self):
                 'test -d /home/user/QubesIncoming/{}'.format(
                     self.testvm1.name)))
 
+    def test_115_qrexec_filecopy_no_agent(self):
+        # The operation should not hang when qrexec-agent is down on target
+        # machine, see QubesOS/qubes-issues#5347.
+
+        self.loop.run_until_complete(asyncio.wait([
+            self.testvm1.start(),
+            self.testvm2.start()]))
+
+        with self.qrexec_policy('qubes.Filecopy', self.testvm1, self.testvm2):
+            # Make sure qrexec-agent is running before killing it
+            self.loop.run_until_complete(asyncio.wait([
+                self.wait_for_session(self.testvm2)]))
```

Comment:
Didn't you wanted to add both testvm1 and testvm2 here? Anyway, qrexec is started as part of `vm.start()` already, wait_for_session is needed only for gui-agent (which I don't think you use here).

Reply:
I had some weird intermittent failures on my machine, might be a local setup issue. I will remove that.